### PR TITLE
Use generic CorrespondenceGenerator interface in front-end for SceneOptimizer

### PR DESCRIPTION
**Pull request by johnwlambert**
Modify several function names of classes from `run()` to `run_*()` for clearer process graph visualization.

### CODE REVIEW

File: gtsfm/frontend/descriptor/convopt.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+wrapping over OpenCV's API.
+
+Note: this is a standalone descriptor.
+
+References:
+https://www.robots.ox.ac.uk/~vedaldi/assets/pubs/simonyan14learning.pdf
+
+Authors: John Lambert
+"""
+import cv2 as cv2
```
Comment (ayushbaid) [inline gtsfm/frontend/descriptor/convopt.py:13]: Unit tests for convopt missing. Actually, can you remove any new module addition from this PR?

File: gtsfm/frontend/inlier_support_processor.py [hunk 2]
```diff
@@ -60,6 +59,20 @@ def run(
...
+            np.array([], dtype=np.uint64),
+            TwoViewEstimationReport(v_corr_idxs=v_corr_idxs, num_inliers_est_model=0),
+        )
+
+        # if (two_view_report.R_error_deg is not None) and (two_view_report.R_error_deg > 5):
+        #     return failure_result
+
+        # if (two_view_report.U_error_deg is not None) and (two_view_report.U_error_deg > 5):
+        #     return failure_result
+
```
Comment (ayushbaid) [inline gtsfm/frontend/inlier_support_processor.py:75]: Why are we removing this?

**Review by ayushbaid — APPROVED** [review]
LGTM overall. Left a few minor comments

File: gtsfm/frontend/inlier_support_processor.py (hunk 2 above)
Comment (johnwlambert) [reply gtsfm/frontend/inlier_support_processor.py]: Oh this code was actually just for debugging (uses GT oracle error). Removed now. ISP is unchanged.

File: gtsfm/frontend/descriptor/convopt.py (hunk 1 above)
Comment (johnwlambert) [reply gtsfm/frontend/descriptor/convopt.py:13]: Sure, I'll remove here, and put in a separate PR.

File: gtsfm/configs/deep_front_end.yaml [hunk 3]
```diff
@@ -5,26 +5,29 @@ SceneOptimizer:
...
   pose_angular_error_thresh: 5 # degrees
 
-  feature_extractor:
-    _target_: gtsfm.feature_extractor.FeatureExtractor
-    detector_descriptor:
-      _target_: gtsfm.frontend.cacher.detector_descriptor_cacher.DetectorDescriptorCacher
-      detector_descriptor_obj:
-        _target_: gtsfm.frontend.detector_descriptor.superpoint.SuperPointDetectorDescriptor
-        max_keypoints: 5000
+  correspondence_generator:
```
[the remark this answers is not in the document]
Comment (johnwlambert) [reply gtsfm/configs/deep_front_end.yaml:8]: Didn't quite catch the last part. We should likely have 3 separate configs now (one for correspondence generator, one for verifier, one for backend). Was that what you were suggesting, or to combine into a mega-config with all 3 parts present?

**Pull request merged**